src/main/java/com/example/bookstorewebapp/service/shoppingcart/impl/ShoppingCartServiceImpl.java

```diff
@@ -0,0 +1,78 @@
+package com.example.bookstorewebapp.service.shoppingcart.impl;
+
+import com.example.bookstorewebapp.dto.cartitem.CreateCartItemRequestDto;
+import com.example.bookstorewebapp.dto.cartitem.UpdateCartItemRequestDto;
+import com.example.bookstorewebapp.dto.shoppingcart.ShoppingCartResponseDto;
+import com.example.bookstorewebapp.mapper.CartItemMapper;
+import com.example.bookstorewebapp.mapper.ShoppingCartMapper;
+import com.example.bookstorewebapp.model.CartItem;
+import com.example.bookstorewebapp.model.ShoppingCart;
+import com.example.bookstorewebapp.model.User;
+import com.example.bookstorewebapp.repository.shoppingcart.ShoppingCartRepository;
+import com.example.bookstorewebapp.service.book.BookService;
+import com.example.bookstorewebapp.service.cartitem.CartItemService;
+import com.example.bookstorewebapp.service.shoppingcart.ShoppingCartService;
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+@RequiredArgsConstructor
+@Service
+public class ShoppingCartServiceImpl implements ShoppingCartService {
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final ShoppingCartMapper cartMapper;
+    private final CartItemService cartItemService;
+    private final CartItemMapper cartItemMapper;
+    private final BookService bookService;
+
+    @Override
+    public void createShoppingCart(User user) {
+        ShoppingCart shoppingCart = new ShoppingCart();
+        shoppingCart.setUser(user);
+        shoppingCartRepository.save(shoppingCart);
+    }
+
+    @Override
+    public ShoppingCartResponseDto getByUserId(Long id) {
+        ShoppingCart shoppingCart = shoppingCartRepository.findShoppingCartByUserId(id);
+        Set<CartItem> cartItems = cartItemService.findAllById(id);
```

Comment:
No reasont to make 2 request, you can fetch from DB shopping card with cart items already